extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java

```diff
@@ -456,15 +464,124 @@ private static Map<String, ParquetTableWriter.GroupingColumnWritingInfo> groupin
         return gcwim;
     }
 
+    /**
+     * Write table to disk in parquet format with {@link TableDefinition#getPartitioningColumns() partitioning columns}
+     * written as "key=value" format in a nested directory structure. To generate these individual partitions, this
+     * method will call {@link Table#partitionBy(String...) partitionBy} on all the partitioning columns.
+     *
+     * @param sourceTable The table to partition and write
+     * @param destinationDir The destination directory to store partitioned data in. Non-existing directories are
+     *        created.
+     * @param baseName The base name for the individual partitioned tables. For example, a base name of "table" will
+     *        result in files named "partition1/table.parquet", "partition2/table.parquet", etc.
+     * @param writeInstructions Write instructions for customizations while writing
+     */
+    public static void writeKeyValuePartitionedTable(@NotNull final Table sourceTable,
+            @NotNull final File destinationDir,
+            @NotNull final String baseName,
+            @NotNull final ParquetInstructions writeInstructions) {
+        // TODO Should I take an additional write instruction for partitioning column names in case they are different
```

Comment:
TODO Confirm with @rcaudy

Reply:
The indexing PR that @rcaudy is working on (#3851) has made a lot of changes to table writing, including the generation of companion index files.  I think this partition-level dividing and writing to disk needs to be harmonized with that PR. 

This looks like good tooling, maybe the indexing PR can/should leverage it to add partition-writing to the other changes.